include/quic/connection.hpp

```diff
@@ -93,63 +143,146 @@ namespace oxen::quic
         Endpoint& endpoint() { return _endpoint; }
         const Endpoint& endpoint() const { return _endpoint; }
 
-      private:
-        std::shared_ptr<ContextBase> context;
-        config_t user_config;
-        Direction dir;
-        Endpoint& _endpoint;
-        const ConnectionID _source_cid;
-        ConnectionID _dest_cid;
-        Path _path;
-        std::function<void(Connection&)> on_closing;  // clear immediately after use
+        int get_streams_available() const override;
+        size_t get_max_datagram_size() const override;
+        int get_max_streams() const override { return _max_streams; }
+        bool datagrams_enabled() const override { return _datagrams_enabled; }
+        bool packet_splitting_enabled() const override { return _packet_splitting; }
+        Splitting packet_splitting_policy() const override { return _policy; }
 
+      private:
         // private Constructor (publicly construct via `make_conn` instead, so that we can properly
         // set up the shared_from_this shenanigans).
         Connection(
                 Endpoint& ep,
                 const ConnectionID& scid,
                 const ConnectionID& dcid,
                 const Path& path,
-                std::shared_ptr<ContextBase> ctx,
-                Direction dir,
+                std::shared_ptr<IOContext> ctx,
                 ngtcp2_pkt_hd* hdr = nullptr);
 
+        ngtcp2_crypto_conn_ref cref;
+        Endpoint& _endpoint;
+        std::shared_ptr<IOContext> context;
+        Direction dir;
+        const ConnectionID _source_cid;
+        ConnectionID _dest_cid;
+        Path _path;
+        std::function<void(Connection&)> on_closing;  // clear immediately after use
+        const int _max_streams{DEFAULT_MAX_BIDI_STREAMS};
+        const bool _datagrams_enabled{false};
+        const bool _packet_splitting{false};
+        const Splitting _policy{Splitting::NONE};
```

Comment:
The variable name `_policy` here seems ambiguous.

Reply:
I dunno. the enum class name is pretty clear what it is, so renaming it to `_splitting_policy` seems redundant. Also it's private and the getter is `Splitting Connection::packet_splitting_policy()`

Reply:
If the getter is used anywhere it is referenced even within this class, that will be fine.  If not, we will later have something like:

```
// user datagram comes in
if (_policy)
{
    // split the datagram if needed
}
else
{
    // don't split
}
```

In that case, reading the code some time in the future it could be confusing/ambiguous what `_policy` means.  It would likely become clear quickly enough by reading the body of the `if` statement, but that feels suboptimal.